jetty-websocket/websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/WebSocketServerComponents.java

```diff
@@ -21,31 +21,50 @@
 import javax.servlet.ServletContext;
 
 import org.eclipse.jetty.io.ByteBufferPool;
+import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.util.DecoratedObjectFactory;
+import org.eclipse.jetty.util.compression.DeflaterPool;
+import org.eclipse.jetty.util.compression.InflaterPool;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
 import org.eclipse.jetty.websocket.core.WebSocketExtensionRegistry;
 
 /**
  * A collection of components which are the resources needed for websockets such as
  * {@link ByteBufferPool}, {@link WebSocketExtensionRegistry}, and {@link DecoratedObjectFactory}.
  *
- * These components should be accessed through {@link WebSocketServerComponents#ensureWebSocketComponents} so that
+ * These components should be accessed through {@link WebSocketServerComponents#getWebSocketComponents} so that
  * the instance can be shared by being stored as a bean on the ContextHandler.
  */
 public class WebSocketServerComponents extends WebSocketComponents
 {
     public static final String WEBSOCKET_COMPONENTS_ATTRIBUTE = WebSocketComponents.class.getName();
 
-    public static WebSocketComponents ensureWebSocketComponents(ServletContext servletContext)
+    WebSocketServerComponents(InflaterPool inflaterPool, DeflaterPool deflaterPool)
     {
-        // Ensure a mapping exists
-        WebSocketComponents components = (WebSocketComponents)servletContext.getAttribute(WEBSOCKET_COMPONENTS_ATTRIBUTE);
+        super(null, null, null, inflaterPool, deflaterPool);
+    }
+
+    public static WebSocketComponents ensureWebSocketComponents(Server server, ServletContext servletContext)
+    {
+        WebSocketComponents components = server.getBean(WebSocketComponents.class);
         if (components == null)
         {
-            components = new WebSocketServerComponents();
-            servletContext.setAttribute(WEBSOCKET_COMPONENTS_ATTRIBUTE, components);
+            InflaterPool inflaterPool = InflaterPool.ensurePool(server);
```

Comment:
IT may be good to have an override mechanism, so you can pass in a non shared pool if you want an only iff null do you use the ensure pool methods